### PR TITLE
add rartland and via support

### PR DESCRIPTION
**Pull request by alabahuy**
## Description

add new keyboard rartland and via support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — APPROVED** [review]

File: keyboards/rart/rartland/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            oled_write_ln_P(PSTR("Undefined\n"), false);
+    }
+
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+}
+#endif
```
Comment (drashna) [inline keyboards/rart/rartland/keymaps/default/keymap.c:89]: could these be moved to the keyboard's c file?

Comment (alabahuy) [reply keyboards/rart/rartland/keymaps/default/keymap.c:89]: @drashna I think that oled and encoder are optional, so leave them in keymap.c

Comment (drashna) [reply keyboards/rart/rartland/keymaps/default/keymap.c:89]: Either way, they should be moved there. 

There are a few reasons for this, but mostly, that it allows for a default to be set that doesn't require a user to explicitly set them up.  This is ESPECIALLY useful for the QMK Configurator, as it means that they will work there.   Having them only in the keymaps means that you cannot use the configurator to get them working.

Comment (alabahuy) [reply keyboards/rart/rartland/keymaps/default/keymap.c:89]: @drashna ok, i will move it to c file, thanks you

File: keyboards/rart/rartland/rartland.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rartland.h"
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    switch (index) {
```
Comment (drashna) [inline keyboards/rart/rartland/rartland.c:19]:
```suggestion
    if (!encoder_update_user(index, clockwise)) { return false; }
    switch (index) {
```

File: keyboards/rart/rartland/rartland.c [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                tap_code(KC_VOLD);
+            }
+        break;
+    }
+    return true;
+}
+#endif
+
+#ifdef OLED_DRIVER_ENABLE
+void oled_task_user(void) {
```
Comment (drashna) [inline keyboards/rart/rartland/rartland.c:33]:
```suggestion
__attribute__((weak)) void oled_task_user(void) {
```

**Review by drashna — CHANGES_REQUESTED** [review]
Just a couple of small things.

File: keyboards/rart/rartland/rartland.c (hunk 3 above)
Comment (alabahuy) [reply keyboards/rart/rartland/rartland.c]: done @drashna :)

**Review by drashna — APPROVED** [review]
Awesome, and thanks!

**Comment by Suschman** [comment]
Layer enumeration in default/keymap.c and via/keymap.c seems to be non-ascending?

**Comment by alabahuy** [comment]
@Suschman OMG, very well thanks. i will fixed it

**Pull request merged**